diagnostics-app/src/app/core/services/theme.service.unit.spec.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Unit Tests for theme.service
+ */
+
+import { TestBed } from '@angular/core/testing';
+import { Theme } from '../enums/global.enums';
+
+import { ThemeService } from './theme.service';
+
+describe('unit: service: theme', () => {
+  let service: ThemeService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({});
+    service = TestBed.inject(ThemeService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+    // default mode
+    expect(service.theme).toBe(Theme.DARK);
+  });
+
+  it('should switch from dark to light', () => {
+    let subscription = service.subscribeOnThemeChange((next: Theme) => {
+      expect(next).toBe(Theme.LIGHT);
+    });
+    service.toggleTheme();
+    expect(service.theme).toBe(Theme.LIGHT);
+    subscription.unsubscribe();
+  });
+
+  it('should switch from dark to light', () => {
+    // switch from dark to light before subscribing
+    service.toggleTheme();
```

Comment:
this is the subject of this test, the test shouldn't depend that it behaves "correctly". Instead, we want to set the initial value of the Theme to be Theme.LIGHT. And this is tricky, because theme is private. One way to do this, is to call themeService constructor with the initial theme value. WDYT?

Reply:
It was indeed tricky to pass a constructor parameter in Angular. But I found a way to do it using Angular's Dependency Injection [[example](https://www.techiediaries.com/angular-10-constructor-parameters-inject-optional/)]. 

Now in future, if needed, `defaultTheme` can be passed to the constructor using the `providers` array in `@NgModule` declaration in `app.module.ts`.

Reply:
Sounds good. Thanks for looking into this.

Using the initial value in the constructor, we can add the feature to detect the system preference and set it accordingly.